library/core/src/main/java/com/google/android/exoplayer2/trackselection/AdaptiveTrackSelection.java

```diff
@@ -540,7 +540,17 @@ public int evaluateQueueSize(long playbackPositionUs, List<? extends MediaChunk>
   @SuppressWarnings("unused")
   protected boolean canSelectFormat(
       Format format, int trackBitrate, float playbackSpeed, long effectiveBitrate) {
-    return Math.round(trackBitrate * playbackSpeed) <= effectiveBitrate;
+
+    boolean isIframeOnly = (format.roleFlags & C.ROLE_FLAG_TRICK_PLAY) != 0;
+    boolean canSelect = Math.round(trackBitrate * playbackSpeed) <= effectiveBitrate;
+
+    if (Math.abs(playbackSpeed) > 6.0f) {
+      canSelect = isIframeOnly;   // TODO factor in playback speed...
```

Comment:
For the TODO, I would change this code to:

```java
    if (Math.abs(playbackSpeed) > 1.0f) {
        canSelect = isIframeOnly && Math.round(trackBitrate * playbackSpeed) <= effectiveBitrate);
    }
```

The other alternative is to modify the track selection's TrackGroups to include only i-Frame tracks on speed change over 1.0.